detect/archive.go

```diff
@@ -0,0 +1,96 @@
+package detect
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mholt/archives"
+	"github.com/zricethezav/gitleaks/v8/sources"
+)
+
+// IsArchive asks archives.Identify (with a nil stream, so only the filename)
+// whether this file would be handled by an Extractor. If Identify returns
+// a Format implementing archives.Extractor, we treat it as an archive.
+func IsArchive(path string) bool {
+	format, _, err := archives.Identify(context.Background(), path, nil)
+	if err != nil {
+		// no matching format at all
+		return false
+	}
+	_, ok := format.(archives.Extractor)
+	return ok
+}
+
+// ExtractArchive extracts all files from archivePath into a temp dir.
+// Returns the list of ScanTargets (with real file paths) and the temp dir for cleanup.
+func ExtractArchive(archivePath string) ([]sources.ScanTarget, string, error) {
+	tmpDir, err := os.MkdirTemp("", "gitleaks-archive-")
```

Comment:
I'm not a fan of how TH fills up the top-level tmp directory. Can we create a single temp dir per scan and nest artifacts under it? e.g.,
```
tmp/
├─ gitleaks-scan-123456/
│  ├─ archive-1234/
│  ├─ archive-4567/
│  ├─ blob.xyz
```